src/core/client_channel/subchannel.cc

```diff
@@ -741,6 +749,7 @@ void Subchannel::StartConnectingLocked() {
   args.channel_args = args_;
   WeakRef(DEBUG_LOCATION, "Connect").release();  // Ref held by callback.
   connector_->Connect(args, &connecting_result_, &on_connecting_finished_);
+  args_ = args_.Remove(GRPC_ARG_SUBCHANNEL_ENDPOINT);
```

Comment:
I don't think we need this anymore, now that we're using `RefCountedPtr<>` to store the endpoint in channel args.